### DISCUSSION TITLE
could desuggar bitstring syntax to anything left to implement on the javascript backend?

### DISCUSSION THREAD

**Discussion by sdancer** [gleam-lang/gleam#1762] [Ideas & suggestions]
could desuggar bitstring syntax to anything left to implement on the javascript backend?

**Comment by lpil** [comment]
Sorry @lissana , I'm not sure what you mean. Could you expand upon that?

**Comment by sdancer** [comment]
<<a:32-little, b, c:binary>>
output some imaginary javascript like calling a bitstring read constructor, then calling imaginary functions on it based on the format
stuff like (reader.read_int_little(32), reader.read_int_little(8), reader.read_binary("rest"))

**Reply by lpil** [reply]
Ah yes! It would be great to support more of the bitstring syntax in JavaScript. Exactly what the API should be isn't clear to me now, but in future having full coverage would be fab.
